repose-aggregator/core/core-lib/src/main/java/org/openrepose/powerfilter/PowerFilterChain.java

```diff
@@ -100,8 +107,35 @@ public void startFilterChain(ServletRequest servletRequest, ServletResponse serv
         servletRequest.setAttribute("filterChainAvailableForRequest", filterChainAvailable);
         servletRequest.setAttribute("http://openrepose.org/requestUrl", ((HttpServletRequest) servletRequest).getRequestURL().toString());
         servletRequest.setAttribute("http://openrepose.org/queryParams", servletRequest.getParameterMap());
+        MutableHttpServletRequest wrappedRequest = MutableHttpServletRequest.wrap((HttpServletRequest) servletRequest);
+        splitRequestHeaders(wrappedRequest);
 
-        doFilter(servletRequest, servletResponse);
+        doFilter(wrappedRequest, servletResponse);
+    }
+
+    private void splitRequestHeaders(MutableHttpServletRequest request) {
```

Comment:
i'd really rather not, exposing that functionality is just giving more rope in to what is already an insanely complicated thing.